src/Features/CSharp/Portable/EditAndContinue/CSharpEditAndContinueAnalyzer.cs

```diff
@@ -1139,6 +1139,36 @@ internal override bool QueryClauseLambdasTypeEquivalent(SemanticModel oldModel,
             }
         }
 
+        protected override void ReportMethodBodySyntaxRudeEditsForLambda(SyntaxNode oldLambda, SyntaxNode newLambda, Match<SyntaxNode> bodyMatch, List<RudeEditDiagnostic> diagnostics)
+        {
+            // We're only interested in local functions here
+            if (!IsLocalFunction(oldLambda))
+            {
+                return;
+            }
+
+            var bodyEditsForLambda = bodyMatch.GetTreeEdits();
```

Comment:
I don't like this. Ideally this would go in the caller, but there is an issue with this where some of our test cases fail when trying to get the tree edits, so I'm guessing there is a bug in the statement syntax comparer, but I haven't dug into it. The `if (!IsLocalFunction..)` is preventing that test failure for now.

Reply:
What tests are failing?

---
In reply to: [557054404](https://github.com/dotnet/roslyn/pull/49027#discussion_r557054404) [](ancestors = 557054404)

Reply:
There are 5 that fail without the if statement, from asserts in StatementSyntaxComparer, for example http://sourceroslyn.io/#Microsoft.CodeAnalysis.CSharp.Features/EditAndContinue/StatementSyntaxComparer.cs,545 throws when it seems a EqualsValueSyntax node. The comment above that assert explains it though - it seems the statement syntax comparer is not necessarily written to be called arbitrarily.

Reply:
I'll take a look.